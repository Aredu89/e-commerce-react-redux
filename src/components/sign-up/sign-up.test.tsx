import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SignUp from './sign-up.component';
import { signUpStart } from '../../store/user/user.action';

const mockedDispatch = jest.fn();
const mockedSetState = jest.fn();
const mockedUseState = ['', mockedSetState];

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: () => mockedUseState
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockedDispatch
}));

describe('Sign-in component', () => {
  it('renders properly', () => {
    const result = render(<SignUp />);
    expect(result).toMatchSnapshot();
  });
  it('captures email input', () => {
    render(<SignUp />);
    
    const input = screen.getByTestId('email-input');

    userEvent.type(input, 'hello');

    expect(mockedSetState).toHaveBeenCalledTimes(5);
    expect(mockedSetState).toHaveBeenCalledWith({"email": "h"});
    expect(mockedSetState).toHaveBeenCalledWith({"email": "he"});
    expect(mockedSetState).toHaveBeenCalledWith({"email": "hel"});
    expect(mockedSetState).toHaveBeenCalledWith({"email": "hell"});
    expect(mockedSetState).toHaveBeenCalledWith({"email": "hello"});
  });
  it('captures password input', () => {
    render(<SignUp />);
    
    const input = screen.getByTestId('password-input');

    userEvent.type(input, 'pass');
    expect(mockedSetState).toHaveBeenCalledTimes(4);
    expect(mockedSetState).toHaveBeenCalledWith({"password": "p"});
    expect(mockedSetState).toHaveBeenCalledWith({"password": "pa"});
    expect(mockedSetState).toHaveBeenCalledWith({"password": "pas"});
    expect(mockedSetState).toHaveBeenCalledWith({"password": "pass"});
  });
  it('logs google user', () => {
    render(<SignUp />);
    const button = screen.getByText('SIGN UP');

    userEvent.click(button);
    const und = undefined as unknown as string;
    expect(mockedDispatch).toHaveBeenCalledWith(signUpStart(und, und, und));
  });
});
