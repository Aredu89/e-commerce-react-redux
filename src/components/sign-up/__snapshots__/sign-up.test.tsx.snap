// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign-in component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-papXJ hGtMkz"
      >
        <h2>
          I do not have an account
        </h2>
        <span>
          Sign up with your email and password
        </span>
        <form
          class="sign-up-form"
        >
          <div
            class="sc-ftvSup cduAcl"
          >
            <input
              class="sc-iBkjds eZAXkO"
              name="displayName"
              required=""
              type="text"
              value=""
            />
            <label
              class="sc-gKXOVf gjofEz"
            >
              Display Name
            </label>
          </div>
          <div
            class="sc-ftvSup cduAcl"
          >
            <input
              class="sc-iBkjds eZAXkO"
              data-testid="email-input"
              name="email"
              required=""
              type="email"
              value=""
            />
            <label
              class="sc-gKXOVf gjofEz"
            >
              Email
            </label>
          </div>
          <div
            class="sc-ftvSup cduAcl"
          >
            <input
              class="sc-iBkjds eZAXkO"
              data-testid="password-input"
              name="password"
              required=""
              type="password"
              value=""
            />
            <label
              class="sc-gKXOVf gjofEz"
            >
              Password
            </label>
          </div>
          <div
            class="sc-ftvSup cduAcl"
          >
            <input
              class="sc-iBkjds eZAXkO"
              name="confirmPassword"
              required=""
              type="password"
              value=""
            />
            <label
              class="sc-gKXOVf gjofEz"
            >
              Confirm Password
            </label>
          </div>
          <button
            class="sc-dkzDqf hdlUww"
            type="submit"
          >
            SIGN UP
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-papXJ hGtMkz"
    >
      <h2>
        I do not have an account
      </h2>
      <span>
        Sign up with your email and password
      </span>
      <form
        class="sign-up-form"
      >
        <div
          class="sc-ftvSup cduAcl"
        >
          <input
            class="sc-iBkjds eZAXkO"
            name="displayName"
            required=""
            type="text"
            value=""
          />
          <label
            class="sc-gKXOVf gjofEz"
          >
            Display Name
          </label>
        </div>
        <div
          class="sc-ftvSup cduAcl"
        >
          <input
            class="sc-iBkjds eZAXkO"
            data-testid="email-input"
            name="email"
            required=""
            type="email"
            value=""
          />
          <label
            class="sc-gKXOVf gjofEz"
          >
            Email
          </label>
        </div>
        <div
          class="sc-ftvSup cduAcl"
        >
          <input
            class="sc-iBkjds eZAXkO"
            data-testid="password-input"
            name="password"
            required=""
            type="password"
            value=""
          />
          <label
            class="sc-gKXOVf gjofEz"
          >
            Password
          </label>
        </div>
        <div
          class="sc-ftvSup cduAcl"
        >
          <input
            class="sc-iBkjds eZAXkO"
            name="confirmPassword"
            required=""
            type="password"
            value=""
          />
          <label
            class="sc-gKXOVf gjofEz"
          >
            Confirm Password
          </label>
        </div>
        <button
          class="sc-dkzDqf hdlUww"
          type="submit"
        >
          SIGN UP
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
